W-18385652 New Auth Method (#398)
mulesoft/docs-access-management: modules/ROOT/pages/configure-client-management-openid-task.adoc

. In *Authentication Method*, complete the information for the type of authentication to use for this client provider:
* *Authorization Header*
+
The HTTP authorization header provides a static token that doesn't expire to authenticate with a server. This header is required if the provider restricts registration requests to authorized clients. If you prefer to use short-lived bearer access tokens, use the OAuth 2.0 private key option instead. 
+
[%autowidth.spread]
|===
|Client Provider | Value
| Okta | `SSWS ${api_token}`, where `api_token` is an API token created through Okta.
| ForgeRock | `Bearer ${api_token}`, where `api_token` is an API token created through ForgeRock.
| Salesforce | `Bearer ${api_token}`, where `api_token` is an API token created through Salesforce.
|===
.. Click *Use OAuth2 private key jwt* to use a private key instead of and authorization header, and provide these values:
* *Token Client ID*
+

> This is the JWT (JSON Web Token) that contains information about the authenticated user, used by OIDC to verify that the user is who they claim to be. 

Reviewer: Hmm, the JWT wouldn't have the authenticated user. Actually, this is no user involved. The JWT is going to be signed by the private key and then the signed JWT is used as a client assertion when Anypoint requests an access token against IDP's `/token` endpoint.  Could include ^ the above context into some doc wording? 

We can borrow Okta's doc here: https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/#use-the-client-credentials-grant-flow

* *Token Client Private Key*
+

> This is the secret cryptographic key used by the client to authenticate itself to a server when requesting access tokens.

Reviewer: `.. used by the client ..` here could be a little confusing. I can foresee support engineers coming at me hah. The key is effectively used by Anypoint to sign the JWT. When IdP (e.g. Okta) reads the signed JWT and recognizes the signature, IdP issues a token to Anypoint. (Then Anypoint uses that token to call /register endpoint to create the client)

* *Token Client API Scopes*
+

> The client API scopes define the specific permissions granted to an application when requesting access to a resource.

Reviewer: `..an application..` The application is actually the token client above. Anypoint uses this token client with the scopes to register new clients.

. In *Authentication Method*, complete the information for the type of authentication to use for this client provider:
* *Authorization Header*
+
The HTTP authorization header provides a static token that doesn't expire to authenticate with a server. This header is required if the provider restricts registration requests to authorized clients. If you prefer to use short-lived bearer access tokens, use the OAuth 2.0 private key option instead. 
+
[%autowidth.spread]
|===
|Client Provider | Value
| Okta | `SSWS ${api_token}`, where `api_token` is an API token created through Okta.
| ForgeRock | `Bearer ${api_token}`, where `api_token` is an API token created through ForgeRock.
| Salesforce | `Bearer ${api_token}`, where `api_token` is an API token created through Salesforce.
|===

> .. Select *Use OAuth2 private key jwt* to use a private key instead of and authorization header, and provide these values:

Reviewer 2: ```suggestion
.. Select *Use OAuth2 private key jwt* to use a private key instead of an authorization header, and provide these values:
```

* *Token Client ID*
+

> Create a JWT (JSON Web Token) and sign it using the private key. Use this as the client assertion when making the `/token` endpoint API call. 

Reviewer: ah my bad! I only focused on JWT thing, but this field isn't about the JWT. Sorry!

```suggestion
This is the unique identifier of the application (i.e. client ID) registered within your IDP. It is used to identify the calling application when requesting a bearer token from the Anypoint platform during client registration. 

```

* *Token Client Private Key*
+

> This is the secret cryptographic key used by Anypoint Platform to sign the JWT. When the IdP reads the signed JWT and recognizes the signature, the IdP issue a token to Anypoint Platform. Anypoint Platform then uses that token to register the endpoint to create the client.

Reviewer: ```suggestion

This is the cryptographic key in PEM format used by Anypoint Platform to sign the JWT for authenticating with the IdP. The corresponding public key must be stored in the IDP in advance. When the IdP reads the signed JWT and validates the signature, the IdP issues an access token used in order for Anypoint Platform to register a new client in the IdP.

```
